mlflow/genai/judges/optimizers/dspy_utils.py

```diff
@@ -0,0 +1,140 @@
+"""Utility functions for DSPy-based alignment optimizers."""
+
+import logging
+from typing import Any
+
+from mlflow.entities.trace import Trace
+from mlflow.exceptions import MlflowException
+from mlflow.genai.judges.judge_trace_utils import (
+    extract_request_from_trace,
+    extract_response_from_trace,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def trace_to_dspy_example(trace: Trace, judge_name: str) -> Any | None:
```

Comment:
Nit - can we make the type hint for DSPy example more specific? 

```suggestion
def trace_to_dspy_example(trace: Trace, judge_name: str) -> "DSPyExample" | None:
```